src/Microsoft.Health.Fhir.CosmosDb/Registration/FhirServerBuilderCosmosDbRegistrationExtensions.cs

```diff
@@ -181,7 +181,21 @@ private static IFhirServerBuilder AddCosmosDbPersistence(this IFhirServerBuilder
                 .Transient()
                 .AsService<ICollectionDataUpdater>();
 
+            services.Add<ResourceManagerCollectionSetup>()
```

Comment:
Would ResourceManagerCollectionSetup always be invoked?
I think for OSS it should, thinking more in terms of SF and AKS?

Reply:
It is registered as self, but not always resolved. If none of the setup functionality is required then you can register the default:
`services.Add<ICollectionSetup, DefaultCollectionSetup>()`

Reply:
Updated to below. When AllowDatabaseCreation and AllowCollectionSetup are false then it will return the noop provider